src/Component/Configuration/GridConfigurationExtender.php

```diff
@@ -23,6 +23,28 @@ public function extends(array $gridConfiguration, array $parentGridConfiguration
 
         unset($configuration['extends']);
 
+        if(isset($configuration['removals'])) {
+            $this->handleRemovals($configuration, $configuration['removals']);
+            unset($configuration['removals']);
+        }
+
         return $configuration;
     }
+
+    private function handleRemovals(array &$configuration, array $removals): void
+    {
+        foreach ($removals as $type => $name) {
+            if (!is_array($name)) {
+                unset($configuration[$name]);
+
+                continue;
+            }
+
+            if(!isset($configuration[$type])) {
+                continue;
+            }
+
+            $this->handleRemovals($configuration[$type], $name);
```

Comment:
```suggestion
            if (is_array($name) && isset($configuration[$type])) {
                $this->handleRemovals($configuration[$type], $name);

                continue;
            }

            unset($configuration[$name]);
```

Reply:
This does not work, but I simplified the loop a bit and changed the test to be a bit clearer